desktop-widgets/downloadfromdivecomputer.cpp

```diff
@@ -153,9 +153,15 @@ void DownloadFromDCWidget::DC##num##Clicked() \
 	ui.product->setCurrentIndex(ui.product->findText(qPrefDiveComputer::product##num())); \
 	bool isBluetoothDevice = isBluetoothAddress(qPrefDiveComputer::device##num()); \
 	bool isMacOs = QSysInfo::kernelType() == "darwin"; \
+	/* If we have a Bluetooth address, use it instead of rescanning. For \
+	MacOs, save the mount point.  This may be the wrong thing to do, but for \
+	the Catalina 10.15.7 laptop and Oceanic Atom 3.1 I use, the mountpoint \
+	is /dev/tty.usbserial-20030001 and it doesn't change regardless of the \
+	USB port being used.   If it isn't saved, it must be reentered. */ \
+	if (isBluetoothDevice || isMacOs) \
+		ui.device->setCurrentText(qPrefDiveComputer::device##num()); \
```

Comment:
I think it will be better to set the device by index, like how it is done in https://github.com/subsurface/subsurface/pull/4120/files#diff-a0bde534c60cabf020b85f0079e0a8d567f5465514e8a59629b3a50a228d17f9L159 - this way, if the remembered device does not currently exist (i.e. the dive computer is not present / detected) the device will be cleared.

Reply:
The change I made in enableBluetoothMode() depends on the value of ui.device->currentText().    Consequently, it must be set before enableBluetoothMode is called (via ui.bluetoothMode->setChecked().    If bluetoothMode is not checked, enableBluetoothMode sets the device index to -1.   Consequently, either the device index must be set both before and after enableBluetoothMode(), or we need a different flag, such as setting CurrentText, for enableBluetoothMode to skip bluetooth scanning.

Reply:
> The change I made in enableBluetoothMode() depends on the value of ui.device->currentText(). Consequently, it must be set before enableBluetoothMode is called (via ui.bluetoothMode->setChecked().

Fully agreed on this part - my point is just that it is probably a better way to set this indirectly, through `ui.device->setCurrentIndex(deviceIndex(qPrefDiveComputer::device##num()))` (might also need a `QCoreApplication::processEvents()` to get `ui.device->currentText()` to update).

> If bluetoothMode is not checked, enableBluetoothMode sets the device index to -1.

Good point, I had not considered the case of remembering the non-bluetooth devices on MacOS.
I suspect that the right thing there is to add another refinement to `enableBluetoothMode()`, to only set the index to -1 on disabling bluetooth if the selected device is a bluetooth device - equivalent to what you are doing when enabling bluetooth.